build.gradle

```diff
@@ -74,6 +74,40 @@ buildscript {
     }
 }
 
+allprojects {
+    // Default to the apache license
+    project.ext.licenseName = 'The Apache Software License, Version 2.0'
+    project.ext.licenseUrl = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+    publishing {
+        repositories {
+            maven {
+                name = 'staging'
+                url = "${rootProject.buildDir}/local-staging-repo"
+            }
+        }
+        publications {
+            // add license information to generated poms
+            all {
+                pom {
+                    name = 'opensearch-flow-framework'
```

Comment:
You may need to add `groupId = "org.opensearch.plugin"` here, should fix your problem

Reply:
It's [here](https://github.com/opensearch-project/flow-framework/pull/187/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R122) already thought. Do you think I should add on L93 as well?

Reply:
Though it's build successfully but not published at the right location with the above change.
```
> Task :publishPluginZipPublicationToMavenLocal
Multiple publications with coordinates 'org.opensearch.plugin:opensearch-flow-framework:3.0.0.0-SNAPSHOT' are published to repository 'mavenLocal'. The publications 'nebula' in root project 'opensearch-flow-framework' and 'pluginZip' in root project 'opensearch-flow-framework' will overwrite each other!

BUILD SUCCESSFUL in 688ms
```

```
ls ~/.m2/repository/org/opensearch/opensearch-
opensearch-benchmarks/                                   opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha256  opensearch-ml-client/
opensearch-cli/                                          opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha512  opensearch-ml-common/
opensearch-common/                                       opensearch-job-scheduler-spi-1.2.0.0-sources.jar         opensearch-ml-spi/
opensearch-compress/                                     opensearch-job-scheduler-spi-1.2.0.0-sources.jar.md5     opensearch-nio/
opensearch-core/                                         opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha1    opensearch-performance-analyzer/
opensearch-dissect/                                      opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha256  opensearch-plugin-classloader/
opensearch-geo/                                          opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha512  opensearch-plugin-cli/
opensearch-grok/                                         opensearch-job-scheduler-spi-1.2.0.0.jar                 opensearch-secure-sm/
opensearch-job-scheduler/                                opensearch-job-scheduler-spi-1.2.0.0.jar.md5             opensearch-ssl-config/
opensearch-job-scheduler-sample-extension/               opensearch-job-scheduler-spi-1.2.0.0.jar.sha1            opensearch-telemetry/
opensearch-job-scheduler-spi/                            opensearch-job-scheduler-spi-1.2.0.0.jar.sha256          opensearch-upgrade-cli/
opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar         opensearch-job-scheduler-spi-1.2.0.0.jar.sha512          opensearch-x-content/
opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.md5     opensearch-launchers/                                    
opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha1    opensearch-ml/  
```

Reply:
> Though it's build successfully but not published at the right location with the above change.

Oh wait, this is for jar publications right?

Reply:
This is for `./gradlew publishToMavenLocal`

Reply:
> This is for `./gradlew publishToMavenLocal`

Thanks @owaiskazi19  , back on that, could you please just clarify one question for me: this plugin should only publish the plugin ZIP or also client JARs?

Reply:
Yes, it should publish the plugin ZIP.